src/spend/stealth.rs

```diff
@@ -0,0 +1,147 @@
+use crate::{decode::decode, Error, JubJubAffine, JubJubExtended};
+
+use std::convert::{TryFrom, TryInto};
+use std::fmt;
+use subtle::{Choice, ConstantTimeEq};
+
+/// The trait `Ownable` is required by any type that wants to prove its
+/// ownership.
+pub trait Ownable {
+    /// Returns the associated `StealthAddress`
+    fn stealth_address<'a>(&'a self) -> &'a StealthAddress;
+}
+
+//. To obfuscate the identity of the participants, we utilizes a Stealth Address
+//. system.
+/// A `StealthAddress` is composed by a one-time public key (`pk_r`, the actual
+// address) and a random point `R`.
+#[derive(Debug, Clone, Copy)]
+pub struct StealthAddress {
+    pub(crate) R: JubJubExtended,
```

Comment:
Why public for the crate if we have the getters?

Reply:
Just to ensure the optimization of the crate, but it's indeed a bit of overkill. I can remove it.

Reply:
Actually no, I forgot why I added in the first place: because currently (by design) it shouldn't be possible instantiate directly `StealthAddress` from outside the `dusk-pki` crates. That's why the visibility `pub(crate)`. Since also there are no other "constructor" (such `new`), the only way to have a `StealthAddress` is to use `gen_stealth_address` from a `PublicSpendKey`.